stats.go

```diff
@@ -536,7 +536,7 @@ func newTypedValue(vd *view.View, r *view.Row) *monitoringpb.TypedValue {
 }
 
 func shouldInsertZeroBound(bounds ...float64) bool {
-	if len(bounds) > 0 && bounds[0] != 0.0 {
+	if len(bounds) > 0 && bounds[0] > 0.0 {
```

Comment:
I digged a bit into the history behind this condition, and it seems like in OpenCensus the negative bounds were explicitly forbidden, see https://github.com/census-instrumentation/opencensus-go/issues/963

You supposedly have encountered this issue in the OpenTelemetry Collector pipeline? Are negative bounds allowed there?

This specific condition was discussed in the original PR, see https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/pull/89#discussion_r260388479 

@rghetia @mayurkale22 do you happen to know more context on this, and possible difference between OpenCensus vs OpenTelemetry in that regard?

Reply:
I came across this when trying to export Prometheus metrics (using OpenTelemetry Prometheus receiver) to Stackdriver (using Stackdriver exporter). The error I got came from Stackdriver Monitoring API, essentially saying that bounds[0] > bounds[1]. So that distribution came all the way through OpenTelemetry just fine:

```
rpc error: code = InvalidArgument desc = Field timeSeries[35].points[0].distributionValue had an invalid value: Distribution |explicit_buckets.bounds| entry 1 has a value of -4.94065645841247e-324 which is less than the value of entry 0 which is 0.
```

Reply:
Interesting. @mayurkale22 claimed in https://github.com/census-instrumentation/opencensus-go/issues/963 that both Prometheus and Stackdriver don't support negative bounds... Has that changed?

Either way, this seems like a harmless change.

If it turns out that Stackdriver indeed doesn't support negative values, we should explicitly validate the buckets in the exporter rather than relying on the OpenCensus / OpenTelemetry code being correct, i.e. fail earlier before invoking this function.

Reply:
So from I have seen I'm pretty sure Prometheus client libs have no problem exposing negative bounds. I'm not actually sure about Stackdriver, it is possible that this change will only turn one error into another. I'll check that.

Reply:
The docs only seem to require that bucket bounds are increasing, nothing about forbidding negative buckets: https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TypedValue#explicit

Reply:
My understanding was that OpenCensus supports only [positive values](https://github.com/census-instrumentation/opencensus-proto/blob/be218fb6bd674af7519b1850cdf8410d8cbd48e8/src/opencensus/proto/metrics/v1/metrics.proto#L213-L215) for distributions. While stackdriver supports [negative values](https://github.com/googleapis/googleapis/blob/40a354b0f47454be4986f9db504cca4d4371c8c3/google/api/distribution.proto#L53-L77) as well. 
So if we convert OC metrics [0, 10), [10, infinity) to Stackdriver (-infinity, 10), [10, infinity) then that would be incorrect. This is because there was no negative number in OC. Hence a zero bound is added with a value of zero. So we have
(-infinity, 0), [0, 10), [10, infinity)

The problem here is that OC rejects -ve bounds. So the SD exporter has to conditionally inject zero bound if the source of the distribution does not allow -ve bound. The SD exporter only accepts [Metricsdata](https://github.com/census-instrumentation/opencensus-go/blob/master/metric/metricdata) or [MetricsProto](https://github.com/census-instrumentation/opencensus-proto/blob/be218fb6bd674af7519b1850cdf8410d8cbd48e8/src/opencensus/proto/metrics/v1/metrics.proto) both of which do not allow -ve bounds. So sending -ve bound would require modification to these data format.

Reply:
So you're saying the bug is in OpenTelemetry exporter which wraps this OCexporter, because it shouldn't have received negative value in the first place? Is OpenTelemetry data model incompatible with OC because it allows negative bounds?

Reply:
Actually opentelemetry-proto also doesn't support [negative bounds](https://github.com/open-telemetry/opentelemetry-proto/blob/b54688569186e0b862bf7462a983ccf2c50c0547/opentelemetry/proto/metrics/v1/metrics.proto#L362-L385). 
/cc @bogdandrutu

Reply:
Interesting. So, next culprit: Prometheus receiver. I saw negative bounds on Prometheus endpoint and that was ingested into OpenTelemetry Collector via Prometheus receiver just fine. Should the negative bound be dropped there instead?

Reply:
If the -ve bound is a legitimate use case then we have to address that in all relevant libraries. If it is not then it is okay to drop it OR with an optional flag allow -ve bounds.

Reply:
I think the right thing to do is to allow negative bounds, they are not really useful for things like request latencies, but may have other use cases (e.g. [Prometheus docs give temperatures as an example](https://prometheus.io/docs/practices/histograms/#count-and-sum-of-observations)).

I just created https://github.com/open-telemetry/opentelemetry-proto/issues/156 to extend the histogram support to negative bucket bounds.

Can this change be merged though? If OT won't end up extending histogram support to cover negative values, this change should be a no-op. If it does, this change will make that compatible with the updated model.

Reply:
I am fine with merging this change. However it will still be incorrect in the following case.
"If the distribution has lowest bound >0 AND if it includes -ve values then this change distorts the distribution by adding additional '0' bound."
May be add this to the comment and link to the above issue.